src/plugins/Network.js

```diff
@@ -374,52 +350,73 @@ module.exports = class Network extends OverlayPlugin {
             const json = {channel: 'network_profile', messages: msgs};
             this.instance.sendEvent(json);
         }
+    }
 
-        const msgs = [];
-        if (this.basebandEnabled) {
-            if (this.networkOperatorMMC.value) {
-                msgs.push('network operator ' + this.networkOperatorMMC.value);
-            }
-            if (this.networkOperatorName.value) {
-                msgs.push('network operator_name ' + this.networkOperatorName.value);
-            }
-
-            if (this.simOperatorMMC.value) {
-                msgs.push('sim operator ' + this.simOperatorMMC.value);
-            }
+    /**
+     * Update form according to the selected profile.
+     */
+    changeMobileProfile() {
+        const profile = MOBILE_PROFILES.find((elem) => elem.name === this.select.value);
+        if (profile) {
+            const msgs = [];
+            msgs.push('setprofile mobile ' + profile.name);
+            const json = {channel: 'network_profile', messages: msgs};
+            this.instance.sendEvent(json);
+        } else {
+            // TODO update profile not found
+            console.log("Selected profile not found")
```

Comment:
Will the TODOs be addressed in this PR? :thinking: (There are a few of them)

Reply:
They are already addressed we just forget ro remove the todo statement :D